geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -217,9 +217,13 @@ impl GeyserPluginManager {
     }
 
     fn _drop_plugin(&mut self, idx: usize) {
+        let current_lib = self.libs.remove(idx);
         let mut current_plugin = self.plugins.remove(idx);
-        let _current_lib = self.libs.remove(idx);
+        let name = current_plugin.name().to_string();
         current_plugin.on_unload();
+        drop(current_plugin);
+        drop(current_lib);
```

Comment:
Just to make sure my understanding is correct, would it also resolve the crash if this line weren't here?

Reply:
Yes, according to https://doc.rust-lang.org/reference/destructors.html#drop-scopes. The variables/temporaries are dropped in the reverse order of the declaration or creation. I intentionally put these explicit drop here to ensure the plugin is dropped before the library.

Reply:
Right, but the library would be dropped automatically at the end of the block.
Anyway, I don't mind the verbosity.

Reply:
there should probably be a comment in the code that this is intentional

Reply:
Will do